File: Task_1/task_1.rs
use std::io::{self, BufRead};

fn is_cyclic_shift(s: &str, t: &str) -> bool {
    if s.len() != t.len() { // Проверяем, равны ли длины строк
        return false;
    }

    let doubled_s = format!("{}{}", s, s); // Создаем строку S+S

    doubled_s.contains(t) // Проверяем, содержится ли T в S+S
}

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    println!("Введите строку S:"); // Считываем первую строку S
    let s = match lines.next() {
        Some(Ok(line)) => line,
        _   {
            println!("Ошибка при вводе строки S.");
            return;
        }
    };

    println!("Введите строку T:"); // Считываем вторую строку T
    let t = match lines.next() {
        Some(Ok(line)) => line,
        _ => {
            println!("Ошибка при вводе строки T.");
            return;
        }
    };

    if is_cyclic_shift(&s, &t) { // Проверяем, является ли T циклическим сдвигом S
        println!("Yes");
    } else {
        println!("No");
    }
}

File: Task_1/task_1.py
def is_cyclic_shift(s: str, t: str) -> bool:
    # Проверяем, равны ли длины строк
    if len(s) != len(t):
        return False

    # Создаем строку S+S
    doubled_s = s + s

    # Проверяем, содержится ли T в S+S
    return t in doubled_s


def main():
    # Считываем строки S и T с клавиатуры
    s = input("Введите строку S: ").strip()
    if not s:
        print("Ошибка при вводе строки S.")
        return

    t = input("Введите строку T: ").strip()
    if not t:
        print("Ошибка при вводе строки T.")
        return

    # Проверяем, является ли T циклическим сдвигом S
    if is_cyclic_shift(s, t):
        print("Yes")
    else:
        print("No")


# Запускаем программу
if __name__ == "__main__":
    main()